store/store.go

```diff
@@ -356,15 +369,27 @@ func (s *Store[H]) Append(ctx context.Context, headers ...H) error {
 		verified, head = append(verified, h), h
 	}
 
+	onWrite := func() {
```

Comment:
Optional. Would name the func to idicate **what** it does, instead of **when**. Like: "storeHead"

Reply:
I made it on purpose. The code in the func should be read as its a regular code and *when* it happens is what matters